### PR TITLE
Configurable actor reminder storage partitions

### PR DESCRIPTION
**Pull request by berndverst**
# Description

Adds support for configuring the actor reminder storage partitions

## Issue reference

https://github.com/dapr/dapr/issues/2889

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by berndverst** [comment]
@rynowak could you take a look at this? Now that https://github.com/dapr/dapr/pull/3297 has landed this needs to make its way into the next DotNet SDK version.

CC @halspang

File: test/Dapr.Actors.Test/Runtime/ActorRuntimeTests.cs [hunk 1]
```diff
@@ -144,6 +145,51 @@ public async Task TestActorSettings()
 
             element = root.GetProperty("drainRebalancedActors");
             Assert.True(element.GetBoolean());
+
+            try {
+                element = root.GetProperty("remindersStoragePartitions");
```
Comment (rynowak) [inline test/Dapr.Actors.Test/Runtime/ActorRuntimeTests.cs:150]: Use `TryGetProperty` instead

File: test/Dapr.Actors.Test/Runtime/ActorRuntimeTests.cs [hunk 2]
```diff
@@ -144,6 +145,51 @@ public async Task TestActorSettings()
 
             element = root.GetProperty("drainRebalancedActors");
             Assert.True(element.GetBoolean());
+
+            try {
```
Comment (rynowak) [inline test/Dapr.Actors.Test/Runtime/ActorRuntimeTests.cs:149]: C# uses Allman style (braces on separate lines)

Comment (berndverst) [reply test/Dapr.Actors.Test/Runtime/ActorRuntimeTests.cs:149]: Ok will update.

I am at the mercy of linters. And VS Code C# extension didn't catch this either.

**Review by rynowak — APPROVED** [review]

**Pull request merged**